src/TiledArray/math/linalg/non-distributed/heig.h

```diff
@@ -52,12 +52,12 @@ namespace TiledArray::lapack {
  */
 template <typename Array>
 auto heig(const Array& A, TiledRange evec_trange = TiledRange()) {
-  using numeric_type = typename lapack::array_traits<Array>::numeric_type;
+  using numeric_type = typename detail::array_traits<Array>::numeric_type;
   World& world = A.world();
-  auto A_eig = detail::to_eigen(A);
+  auto A_eig = detail::make_matrix(A);
```

Comment:
Since `make_matrix` replicates here, wouldn't it make more sense to add a "scatter/gather" type driver rather than "Alltoall` to save on some communication dependencies?